### PR TITLE
Create separate gateway sub chart

### PR DESCRIPTION
**Pull request by Zelldon**
Create a separate sub chart for the standalone gateway.

 * For better separation of concerns and improving maintainability (makes the charts smaller / simplifies them).
 * It would be easier to disable the gateway (if necessary)
 * In the parent values file we have a new separate section for the gateway (`zeebe-gateway`)

related #124

### CODE REVIEW

File: charts/ccsm-helm/charts/zeebe-gateway/README.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
+[![Community Extension](https://img.shields.io/badge/Community%20Extension-An%20open%20source%20community%20maintained%20project-FF4700)](https://github.com/camunda-community-hub/community)[![Lifecycle: Incubating](https://img.shields.io/badge/Lifecycle-Incubating-blue)](https://github.com/Camunda-Community-Hub/community/blob/main/extension-lifecycle.md#incubating-)[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
```
Comment (Zelldon) [inline charts/ccsm-helm/charts/zeebe-gateway/README.md:1]: This file will be completely replaced later, when I iterate over the readmes.

**Comment by npepinpe** [comment]
Let me know if you need a reviewer additionally, but I assumed one was OK :+1:

File: charts/ccsm-helm/charts/zeebe-gateway/templates/configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  name: {{ include "zeebe-gateway.fullname" . }}
+  labels: 
+    {{- include "zeebe.labels.gateway" . | nindent 4 }}
+apiVersion: v1
+data:
+  startup.sh: |
+    #!/usr/bin/env bash
+    set -eux -o pipefail
+
+    export ZEEBE_BROKER_NETWORK_ADVERTISEDHOST=${ZEEBE_BROKER_NETWORK_ADVERTISEDHOST:-$(hostname -f)}
```
Comment (menski) [inline charts/ccsm-helm/charts/zeebe-gateway/templates/configmap.yaml:12]: will you rework this file later? Not sure if these configs are useful for the gateway?

File: charts/ccsm-helm/charts/zeebe-gateway/templates/_helpers.tpl [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+{{- end -}}
+{{- end -}}
+
+{{- define "zeebe.version" -}}
+{{- printf "%s:%s" .Values.global.image.repository .Values.global.image.tag -}}
+{{- end -}}
+
+{{- define "zeebe.labels.gateway" -}}
+{{- template "zeebe.labels" . }}
+app.kubernetes.io/component: gateway
```
Comment (menski) [inline charts/ccsm-helm/charts/zeebe-gateway/templates/_helpers.tpl:27]:
```suggestion
app.kubernetes.io/component: zeebe-gateway
```
Would this be more consistent?

**Review by menski — COMMENTED** [review]
lgtm to me, just two comments

File: charts/ccsm-helm/charts/zeebe-gateway/templates/configmap.yaml (hunk 2 above)
Comment (Zelldon) [reply charts/ccsm-helm/charts/zeebe-gateway/templates/configmap.yaml:12]: Yes I will work on this. Currently it is the same as before. The Standalone gateway before used the same configmap as the broker. :)

File: charts/ccsm-helm/charts/zeebe-gateway/templates/_helpers.tpl (hunk 3 above)
Comment (Zelldon) [reply charts/ccsm-helm/charts/zeebe-gateway/templates/_helpers.tpl:27]: Not sure, for the brokers we have as label, `component: broker` :shrug: Would you prefer it ? :)

Comment (menski) [reply charts/ccsm-helm/charts/zeebe-gateway/templates/_helpers.tpl:27]: I guess if we not look from a zeebe focused perspective but from a CCSM user perspective, I would expect prefixed labels with the component names, i.e. `zeebe-`, `operate-` etc.

So yes personally I would expect `zeebe-gateway` and `zeebe-broker`

File: charts/ccsm-helm/charts/zeebe-gateway/templates/configmap.yaml (hunk 2 above)
Comment (menski) [reply charts/ccsm-helm/charts/zeebe-gateway/templates/configmap.yaml:12]: okay

File: charts/ccsm-helm/charts/zeebe-gateway/templates/_helpers.tpl (hunk 3 above)
Comment (Zelldon) [reply charts/ccsm-helm/charts/zeebe-gateway/templates/_helpers.tpl:27]: Makes sense to me let me do that as part of https://github.com/camunda-community-hub/camunda-cloud-helm/pull/147

**Comment by Zelldon** [comment]
Thanks @menski for the review :bow:

**Pull request merged**